app/controllers/api/v1/merchants_invoices_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Api::V1::MerchantsInvoicesController < ApplicationController
+    ALLOWED_STATUSES = ["shipped", "returned", "packaged"].freeze
+    #makes an unchangeable array of the allowed statuses for invoices
+
+    def index   # handling 'get' api/v1/merchants/:merchant_id/invoices    
+        status = params[:status]
+
+        #finding the invoices
+        if status.present?
+            invoices = Invoice.where(merchant_id: params[:merchant_id], status: status)
+        else
+            invoices = Invoice.where(merchant_id: params[:merchant_id], status: ALLOWED_STATUSES)
```

Comment:
Something to think about when refactoring would be maybe moving the Active Record logic into the model to keep the controller cleaner and keeping the data logic in the model